digest/set.go

```diff
@@ -63,6 +65,8 @@ func checkShortMatch(alg Algorithm, hex, shortAlg, shortHex string) bool {
 // with an empty digest value. If multiple matches are found
 // ErrDigestAmbiguous will be returned with an empty digest value.
 func (dst *Set) Lookup(d string) (Digest, error) {
+	dst.mutex.RLock()
```

Comment:
This string does not have to be a digest, the value may be `421c76d7` to match digest value `sha256:421c76d77563afa1914846b010bd164f395bd34c2102e5e99e0cb9cf173c1d87`